Create profile - issue #11

woontoon opened the pull request:
checking to see if this is the right direction we want the account creation to go
account create -> /profile/create to link an lw account 
users can edit the rest of their profile whenever they want as only an lw account is required

quinn-dougherty commented on app/api/lesswrong/route.ts:
```diff
@@ -0,0 +1,153 @@
+import { NextRequest } from "next/server";
```
i think endpoint name should be `lesswrongAuth`, right?

quinn-dougherty commented on app/profile/create/page.tsx:
```diff
@@ -0,0 +1,18 @@
+import React from "react";
```
i'm not sure about this whole page-- profile isn't being created here, it's just being populated. the auth and the trigger (copying from `auth.users` table to `public.profiles` table) is what creates the profie. I think this is more like `app/profile/edit/page.tsx`?

quinn-dougherty commented on components/profile/LesswrongLink.tsx:
```diff
@@ -0,0 +1,92 @@
+"use client";
+import React, { useState } from "react";
+
+const LesswrongLink = () => {
```
I think `LesswrongConnect` is a little better, isn't it? cuz `Link` has other meanings, whereas `Connect` sounds closer to auth. you could also call the component `LesswrongAuth`?

quinn-dougherty reviewed (commented):
just some relatively insignificant variable/route name ideas. Seems like the right direction overall!

quinn-dougherty commented:
- [ ] #44 

Can you please piggyback starting people off with 5-10k of playmoney, in `balance`?

quinn-dougherty commented:
Can `app/lesswrong.svg` be `app/img/lesswrong.svg` plz?

woontoon commented:
- added checks for logging in, if you don't already have an lw account linked it should redirect you to /profile/create
- same for signing up
- changed the page for authenticating lw accounts to be the same as the login page
- disabled unlinking/relinking of an lw account
- added checks for creating posts
- assigned 10k in play money upon authenticating with lw

let me know your thoughts

quinn-dougherty commented on app/api/lesswrongAuth/route.ts:
```diff
@@ -0,0 +1,180 @@
+
+const lesswrongEndpoint = "https://www.lesswrong.com/graphql";
+
+interface lesswrongApiResponse<T> {
```
(i'll allow it for code velocity) style throughout codebase has been interfaces in `CamelCase` rather than `camelCase`

quinn-dougherty reviewed (approved):
LGTM, I'll fix `github action` build (it broke because I renamed the project) when I get to wework then I'll merge

quinn-dougherty commented:
re jobs failing: I verified that this isn't a gh action problem on main. idk what's going on. gonna merge in.
